includes/Shopping/Woocommerce_Query.php

```diff
@@ -104,6 +108,20 @@ public function get_search( string $search_term ) {
 		return $results;
 	}
 
+	/**
+	 * Get all product image ids (feature image + gallery_images).
+	 *
+	 * @since 1.21.0
+	 *
+	 * @param \WC_Product $product 
+	 * @return array
+	 */
+	protected function get_product_image_ids( $product ): array {
+		$product_image_ids = array_merge( [ $product->get_image_id() ], $product->get_gallery_image_ids() );
```

Comment:
```suggestion
		$product_image_ids = $product->get_gallery_image_ids();
		array_unshift( $product_image_ids, $product->get_image_id() );
```

Using array_merge in an loop should be avoided. See https://dev.to/klnjmm/never-use-arraymerge-in-a-for-loop-in-php-5go1

We are just adding to the start of an array, unshift should work here.